src/main/java/io/cdap/e2e/pages/actions/CdfPipelineRunAction.java

```diff
@@ -107,10 +122,20 @@ public static void waitForPipelineToTransitionToStatus(String pipelineStatus, lo
 
   /**
    * Wait till the Pipeline's status changes (from Running) to either Succeeded, Failed or Stopped within the
-   * Timeout: {@link ConstantsUtil#PIPELINE_RUN_TIMEOUT_SECONDS}
+   * Timeout: {@link ConstantsUtil#IMPLICIT_TIMEOUT_SECONDS}
    */
-  public static void waitTillPipelineRunCompletes() {
-    SeleniumDriver.getWaitDriver(ConstantsUtil.PIPELINE_RUN_TIMEOUT_SECONDS).until(ExpectedConditions.or(
+  public static void waitTillPipelineRunCompletes() throws IOException {
+    // Adding a page refresh in case tests are running on CDF to update the pipeline status.
+      RetryUtils.retry(ConstantsUtil.PIPELINE_REFRESH_TIMEOUT_SECONDS, ConstantsUtil.PIPELINE_RUN_TIMEOUT_SECONDS,
+        10, () -> {
+        logger.info("Refreshing the page.");
+        PageHelper.refreshCurrentPage();
+        SeleniumDriver.getWaitDriver(ConstantsUtil.SMALL_TIMEOUT_SECONDS);
+        return !(isStarting() || isRunning() || isProvisioning());
+        }
+      );
+
+    SeleniumDriver.getWaitDriver(ConstantsUtil.IMPLICIT_TIMEOUT_SECONDS).until(ExpectedConditions.or(
```

Comment:
This looks similar to `waitTillPipelineRunCompletes`. Please reuse.

Reply:
Re-used (66e7a8071c28508450e1813edfeda2ccb7be0f60)